code/processes/wdb.q

```diff
@@ -300,8 +300,8 @@ endofdaymerge:{[dir;pt;tablist;mergelimits]
 		merge[dir;pt;;mergelimits] peach flip (key tablist;value tablist);];	
 		[.lg.o[`merge;"merging on master"];
 		merge[dir;pt;;mergelimits] each flip (key tablist;value tablist)]];
-	/- delete the empty date directory
-	.os.deldir .os.pth[string .Q.par[savedir;pt;`]];	
+	/- if directory exists delete it
+        $[`error ~ @[get; (hsym `$.os.pth[string .Q.par[savedir;pt;`]]); {`error}]; (); .os.deldir .os.pth[string .Q.par[savedir;pt;`]]];
```

Comment:
You can derive the path just once rather than twice (i.e. this bit: .os.pth[string .Q.par[savedir;pt;`]])

To check existence it is better to use key

key on a directory that exists returns a symbol list of it's contents
key on a file that exists returns the name of the file as a symbol
key on a file/directory that doesn't exist returns empty list ()